lib/user-methods.js

```diff
@@ -62,6 +62,19 @@ function checkObjectTypes(obj, types) {
     }
 }
 
+function normalizeSyncUrl(authUrl, syncUrl) {
+    let parsedAuthUrl = new URL(authUrl);
+    let secure = (parsedAuthUrl.protocol === 'https:') ? 's' : '';
```

Comment:
This seems like an overly clever way to save a few letters 😛I think it'd be much more obvious if we were explicit about what this is and why we need it. Something like:

```js
const realmProtocol = parsedAuthUrl.protocol === "https:" ? "realms" : "realm";
```